kernel/src/vm/mod.rs

```diff
@@ -143,6 +143,9 @@ fn promote_hugepages(proc: &Arc<Process>, addr_hint: Option<Vaddr>) -> Result<()
 
         if (range.end - range.start) >= PROMOTED_PAGE_SIZE {
             // Already huge, nothing to do here
+            if cursor.jump(range.end).is_err() {
+                break;
+            }
```

Comment:
I would prefer `return Ok(())` instead of break because it makes it clear we are aborting the whole function.